cmd/cli/certificate_rotate.go

```diff
@@ -0,0 +1,352 @@
+	if !r.promptForContinue("Are you sure you want to initiate rotation?") {
+		return nil
+	}
+
+	oldMrc, err := r.findCurrentInUse()
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(r.out, "Found MeshRootCertificate [%s] for mesh [%s] in namespace [%s]\n\n", oldMrc.Name, r.meshName, settings.Namespace())
+
+	newMrc, err := r.createOrFindNewMRC(oldMrc)
+	if err != nil {
+		return fmt.Errorf("unable to create or find new MeshRootCertificate: %s", err)
+	}
+	fmt.Fprintf(r.out, "Created new MeshRootCertificate [%s] in Inactive role\n", newMrc.Name)
+
+	fmt.Fprintf(r.out, "Moving MeshRootCertificate [%s] to Passive role\n", newMrc.Name)
+	if !r.promptForContinue("Are you sure you want to initiate rotation?") {
+		return nil
+	}
+	err = r.updateCertificate(newMrc.Name, v1alpha2.PassiveIntent)
+	if err != nil {
+		return fmt.Errorf("unable to update MeshRootCertificate [%s] with role [%s]: %s", newMrc.Name, v1alpha2.PassiveIntent, err)
+	}
+	fmt.Fprintf(r.out, "waiting for propagation...\n\n")
+	time.Sleep(r.waitForRotation)
+
+	fmt.Fprintf(r.out, "Moving MeshRootCertificate [%s] to Active role\n", newMrc.Name)
+	if !r.promptForContinue("Are you sure you want to initiate rotation?") {
+		return nil
+	}
+	err = r.updateCertificate(newMrc.Name, v1alpha2.ActiveIntent)
+	if err != nil {
+		return fmt.Errorf("unable to update MeshRootCertificate [%s] with role [%s]: %s", newMrc.Name, v1alpha2.ActiveIntent, err)
+	}
+	fmt.Fprintf(r.out, "waiting for propagation...\n\n")
+
+	time.Sleep(r.waitForRotation)
+
+	fmt.Fprintf(r.out, "Moving MeshRootCertificate [%s] to Passive role\n", oldMrc.Name)
+	if !r.promptForContinue("Are you sure you want to initiate rotation?") {
+		return nil
+	}
+	err = r.updateCertificate(oldMrc.Name, v1alpha2.PassiveIntent)
+	if err != nil {
+		return fmt.Errorf("unable to update MeshRootCertificate [%s] with role [%s]: %s", oldMrc.Name, v1alpha2.PassiveIntent, err)
+	}
+	fmt.Fprintf(r.out, "waiting for propagation...\n\n")
+	time.Sleep(r.waitForRotation)
+
+	fmt.Fprintf(r.out, "Moving MeshRootCertificate [%s] to Inactive role\n", oldMrc.Name)
+	if !r.promptForContinue("Are you sure you to move the old MeshRootCertificate to Inactive?") {
+		return nil
+	}
+	err = r.updateCertificate(oldMrc.Name, v1alpha2.InactiveIntent)
+	if err != nil {
+		return fmt.Errorf("unable to update MeshRootCertificate [%s] with role [%s]: %s", oldMrc.Name, v1alpha2.InactiveIntent, err)
+	}
+	fmt.Fprintf(r.out, "waiting for propagation...\n\n")
+	time.Sleep(r.waitForRotation)
+
+	if r.deleteOld {
+		if !r.promptForContinue("Are you sure you want to delete the old MeshRootCertificate?") {
+			return nil
+		}
+		err = r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Delete(context.Background(), oldMrc.Name, metav1.DeleteOptions{})
+		if err != nil {
+			fmt.Printf("warning: unable to delete MeshRootCertificate [%s]", oldMrc.Name)
+		}
+		if oldMrc.Spec.Provider.Tresor != nil {
```

Comment:
do we have a protection in place that the old mrc's tresor secret ref isn't the same as the new one?

Reply:
added a check and warning